packages/contracts/contracts/extensions/OrderMatcher/MixinMatchOrders.sol

```diff
@@ -0,0 +1,323 @@
+                    // |---------------------------|---------|-------------------------------------------- |
+                    // | 416                       | 4       | `fillOrder` function selector               |
+                    // |                           | 3 * 32  | function parameters:                        |
+                    // | 420                       |         |   1. offset to rightOrder (*)               |
+                    // | 452                       |         |   2. takerAssetFillAmount                   |
+                    // | 484                       |         |   3. offset to rightSignature (*)           |
+                    // |                           | 12 * 32 | rightOrder:                                 |
+                    // | 516                       |         |   1.  senderAddress                         |
+                    // | 548                       |         |   2.  makerAddress                          |
+                    // | 580                       |         |   3.  takerAddress                          |
+                    // | 612                       |         |   4.  feeRecipientAddress                   |
+                    // | 644                       |         |   5.  makerAssetAmount                      |
+                    // | 676                       |         |   6.  takerAssetAmount                      |
+                    // | 708                       |         |   7.  makerFeeAmount                        |
+                    // | 740                       |         |   8.  takerFeeAmount                        |
+                    // | 772                       |         |   9.  expirationTimeSeconds                 |
+                    // | 804                       |         |   10. salt                                  |
+                    // | 836                       |         |   11. offset to rightMakerAssetData (*)     |
+                    // | 868                       |         |   12. offset to rightTakerAssetData (*)     |
+                    // | 900                       | 32      | rightMakerAssetData Length                  |
+                    // | 932                       | a       | rightMakerAssetData Contents                |
+                    // | 932 + a                   | 32      | rightTakerAssetData Length                  |
+                    // | 964                       | b       | rightTakerAssetData Contents                |
+                    // | 964 + b                   | 32      | rightSigature Length (always 0)             |
+
+                    // We assume that `leftOrder.makerAssetData == rightOrder.takerAssetData` and `leftOrder.takerAssetData == rightOrder.makerAssetData`
+                    // `EXCHANGE.matchOrders` already makes this assumption, so it is likely
+                    // that the `rightMakerAssetData` and `rightTakerAssetData` in calldata are empty
```

Comment:
I tried this, but it ended up being the exact same amount of work since we can't make assumptions about how the calldata is layed out and need to load from the order offsets regardless.